sql_generators/glean_usage/templates/events_stream_v1.query.sql

```diff
@@ -1,8 +1,73 @@
 {{ header }}
+-- convert array of key value pairs to a json object, cast numbers and booleans if possible
+CREATE TEMP FUNCTION from_map_event_extra(input ARRAY<STRUCT<key STRING, value STRING>>)
+RETURNS json AS (
+  IF(
+    ARRAY_LENGTH(input) = 0,
+    NULL,
+    JSON_OBJECT(
+      ARRAY(SELECT key FROM UNNEST(input)),
+      ARRAY(
+        SELECT
+          CASE
+            WHEN SAFE_CAST(value AS NUMERIC) IS NOT NULL
+              THEN TO_JSON(SAFE_CAST(value AS NUMERIC))
+            WHEN SAFE_CAST(value AS BOOL) IS NOT NULL
+              THEN TO_JSON(SAFE_CAST(value AS BOOL))
+            ELSE TO_JSON(value)
+          END
+        FROM
+          UNNEST(input)
+      )
+    )
+  )
+);
```

Comment:
THIS is the bigquery SQL I was looking for. I couldn't get it done, which is the sole reason I wrote that that JS UDF. Glad someone else is better at it then I am.